src/main/java/mods/eln/sixnode/EmergencyLamp.kt

```diff
@@ -0,0 +1,320 @@
+    val mainWallL: Obj3D.Obj3DPart = model.getPart("coreWallL")
+    val lightWall: Obj3D.Obj3DPart = model.getPart("lightWall")
+
+    init {
+        voltageLevelColor = VoltageLevelColor.fromCable(cable)
+        changeDefaultIcon("emergencylamp")
+    }
+
+    fun draw(onCeiling: Boolean = false, on: Boolean = false, mirrorSign: Boolean = false) {
+        if (onCeiling) {
+            mainCeiling.draw()
+
+            if (on)
+                preserveMatrix {
+                    UtilsClient.drawLight(panelCeiling)
+                    GL11.glColor3f(0.3f, 0.3f, 0.3f)
+                    UtilsClient.drawLight(lightCeiling)
+                }
+            else
+                panelCeiling.draw()
+        } else {
+            if (on)
+                preserveMatrix {
+                    UtilsClient.drawLight(mainWall)
+                    UtilsClient.drawLight(if (mirrorSign) mainWallL else mainWallR)
+                    GL11.glColor3f(0.3f, 0.3f, 0.3f)
+                    UtilsClient.drawLight(lightWall)
+                }
+            else
+                preserveMatrix {
+                    mainWall.draw()
+                    (if (mirrorSign) mainWallL else mainWallR).draw()
+                }
+        }
+    }
+
+    override fun getFrontFromPlace(side: Direction?, player: EntityPlayer?)
+        = super.getFrontFromPlace(side, player).inverse()
+
+    override fun canBePlacedOnSide(player: EntityPlayer?, side: Direction?): Boolean {
+        if (side == Direction.YN) {
+            Utils.addChatMessage(player, I18N.tr("Emergency lamp can not be placed on ground."))
+            return false
+        } else
+            return true
+    }
+
+    override fun addInformation(itemStack: ItemStack?, entityPlayer: EntityPlayer?, list: MutableList<String>,
+                                par4: Boolean) {
+        with(list) {
+            add("As long as power is provided, the internal battery")
+            add("is charged and the lamp is off. On a power failure,")
+            add("the lamp turns on and runs on batteries.")
+            add(Utils.plotVolt("Nominal voltage:", cable.electricalNominalVoltage))
+            add(Utils.plotEnergy("Battery capacity:", batteryCapacity))
+        }
+    }
+}
+
+class EmergencyLampElement(sixNode: SixNode, side: Direction, descriptor: SixNodeDescriptor)
+    : SixNodeElement(sixNode, side, descriptor) {
+
+    enum class Event(val value: Byte) {
+        TOGGLE_POWERED_BY_CABLE(1),
+        SET_CHANNEL(2)
+    }
+
+    val desc = descriptor as EmergencyLampDescriptor
+    val load = NbtElectricalLoad("load")
+    val chargingResistor = ResistorSwitch("chargingResistor", load, null)
+    var on = false
+        set(value) {
+            if (field != value) needPublish()
+            field = value
+            sixNode.lightValue = if (value) desc.lightLevel else 0
+        }
+    var charge = desc.batteryCapacity / 2
+    var poweredByCable = false
+        set(value) {
```

Comment:
Ditto... though rather than immediately adding pre-change and post-change parameters, reconnection is common enough that a 'reconnect' optional bool might be appropriate.